### PR TITLE
[PyOV] Make graph tests hardware agnostic - part 1

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Remove inference from graph tests to make them hardware agnostic
 - part 2: https://github.com/openvinotoolkit/openvino/pull/14519

### Tickets:
 - 98666

### CODE REVIEW

File: src/bindings/python/tests/test_graph/test_adaptive_pool.py [hunk 1]
```diff
@@ -1,69 +0,0 @@
-# -*- coding: utf-8 -*-
-# Copyright (C) 2018-2022 Intel Corporation
-# SPDX-License-Identifier: Apache-2.0
-
-import openvino.runtime.opset8 as ov
-import numpy as np
-from tests.runtime import get_runtime
-
-
-def test_adaptive_avg_pool():
```
Comment (p-wysocki) [inline src/bindings/python/tests/test_graph/test_adaptive_pool.py:10]: Tests from this file are already covered in https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_graph/test_create_op.py, both Adaptive Max Pool and Adaptive Average Pool.

File: src/bindings/python/tests/test_graph/test_convolution.py [hunk 2]
```diff
@@ -5,143 +5,66 @@
...
-                    ],
-                ],
-            ],
-            dtype=np.float32,
-        ),
-    )
+    node = ov.convolution(input_x, input_filter, strides, pads_begin, pads_end, dilations)
+    assert node.get_type_name() == "Convolution"
+    assert node.get_output_size() == 1
+    assert list(node.get_output_shape(0)) == expected_shape
```
Comment (mitruska) [inline src/bindings/python/tests/test_graph/test_convolution.py:28]: Testing of `node.get_output_element_type(0)` can be considered as additional assert.

File: src/bindings/python/tests/test_graph/test_convolution.py [hunk 3]
```diff
@@ -5,143 +5,66 @@
 import numpy as np
 
 import openvino.runtime.opset8 as ov
-from tests.runtime import get_runtime
-from tests.test_graph.test_ops import convolution2d
-from tests.test_graph.util import run_op_node
 
 
 def test_convolution_2d():
```
Comment (jiwaszki) [inline src/bindings/python/tests/test_graph/test_convolution.py:10]: Are you able to parametrize this kind of tests? Or at least split test cases?

I see a lot repetitions for the same operator. Also there is a lot of redefinitions like `strides`/`pads_begin` etc.

**Review by jiwaszki — APPROVED** [review]
Overall LGTM, small nit about test-style inherited from previous code.

**Comment by ilya-lavrenov** [comment]
merging first part. Comments can be addressed in the second PR

File: src/bindings/python/tests/test_graph/test_convolution.py (hunk 2 above)
Comment (p-wysocki) [reply src/bindings/python/tests/test_graph/test_convolution.py:28]: I'll add it wherever applicable, as Ilya mentioned - in the second part.

File: src/bindings/python/tests/test_graph/test_convolution.py (hunk 3 above)
Comment (p-wysocki) [reply src/bindings/python/tests/test_graph/test_convolution.py:10]: I'll get to it, let's move the discussion to the second part.

**Pull request merged**